File: api/src/main/java/com/example/athletehub/dto/DayResponse.java
package com.example.athletehub.dto;

import java.time.LocalDate;
import java.util.List;

/**
 * {@code GET /api/diet/day?date=YYYY-MM-DD} payload.
 *
 * <ul>
 *   <li>{@code totals} = sum of scaled macros across {@code entries}.</li>
 *   <li>{@code target} = active plan's daily target, or null when no
 *       plan is active.</li>
 *   <li>{@code remaining} = {@code target − totals}, per macro; null
 *       when {@code target} is null. Values can go negative when the
 *       user has gone over (the chart will show "1200 kcal over").</li>
 * </ul>
 */
public record DayResponse(
        LocalDate date,
        List<DiaryEntryDto> entries,
        Macros totals,
        Macros target,
        Macros remaining
) {}

File: client/lib/models/responses/day_response.dart
import 'diary_entry.dart';
import 'macros.dart';

/// `GET /api/diet/day?date=…` payload.
///
/// * `target` and `remaining` are both null when no active diet plan is
///   set — same nullable-axis rule as the body-fat XOR (AH-041) and the
///   today-plan response (AH-032). The four combinations are all real
///   UX states (no diary + no plan, diary + no plan, no diary + plan,
///   diary + plan).
/// * `remaining` can go negative when the user is over target.
class DayResponse {
  DayResponse({
    required this.date,
    required this.entries,
    required this.totals,
    this.target,
    this.remaining,
  });

  final DateTime date;
  final List<DiaryEntry> entries;
  final Macros totals;
  final Macros? target;
  final Macros? remaining;

  bool get hasTarget => target != null;

  factory DayResponse.fromJson(Map<String, dynamic> json) {
    final raw = (json['entries'] as List<dynamic>? ?? const []);
    return DayResponse(
      date: DateTime.parse(json['date'] as String),
      entries:
          raw.map((e) => DiaryEntry.fromJson(e as Map<String, dynamic>)).toList(),
      totals: Macros.fromJson(json['totals'] as Map<String, dynamic>),
      target: json['target'] == null
          ? null
          : Macros.fromJson(json['target'] as Map<String, dynamic>),
      remaining: json['remaining'] == null
          ? null
          : Macros.fromJson(json['remaining'] as Map<String, dynamic>),
    );
  }
}
